### DISCUSSION TITLE
2023/08/17/parse-dont-validate-and-other-type-safety-driven-shenanigans-plus-a-csharp-wishlist/

### DISCUSSION THREAD

**Discussion by giscus** [CodingMilitia/Blog#96] [Blog comments]
# 2023/08/17/parse-dont-validate-and-other-type-safety-driven-shenanigans-plus-a-csharp-wishlist/

Making use of the type system is something I feel should be important when working in a strongly typed language like C#. However, I don’t feel like that’s the case, and I would love for the language to push folks in the direction of creating more robust programs, where the compiler provides more help in proving the correctness of the code.

https://blog.codingmilitia.com/2023/08/17/parse-dont-validate-and-other-type-safety-driven-shenanigans-plus-a-csharp-wishlist/

**Comment by nikiz0r** [comment]
Awesome post. Got a bit confused with the language-ext part tbh since I hadn't had any experience in the past with anything like that, but it does start looking better the longer you look at it 😁

**Comment by mcintyre321** [comment]
Have you tried my OneOf library? It's a much smaller library than LanguageExt as it's just for unions, but it does support Try methods as well as switch and match

**Reply by joaofbantunes** [reply]
I knew of the library's existence, but haven't had the chance to try it yet.
Maybe it can simplify something, but probably not enough to minimize my C# rant 😅.

**Comment by SeanKilleen** [comment]
Great example! I've been meaning to look into `language-ext` for a bit now and I happen to have a use case right when your post popped onto my radar. Thanks!

**Reply by joaofbantunes** [reply]
Glad to hear.
I don't feel like I did the greatest of advertisements to `language-ext`, but it is a pretty cool library!
